src/receive_arbiter.cc

```diff
@@ -0,0 +1,256 @@
+	} else {
+		mrt = &m_transfers[trid].emplace<multi_region_transfer>(elem_size);
+	}
+
+	// Add a new region_request to the `mrt` (transfers have transfer_id granularity, but there might be multiple receives from independent range mappers
+	assert(std::all_of(mrt->active_requests.begin(), mrt->active_requests.end(),
+	    [&](const stable_region_request& rr) { return region_intersection(rr->incomplete_region, request).empty(); }));
+	auto& rr = mrt->active_requests.emplace_back(std::make_shared<region_request>(request, allocation, allocated_box));
+
+	// If the new region_request matches any of the still-unassigned pilots associated with `mrt`, immediately initiate the appropriate payload-receives
+	const auto assign_pilot = [&](const inbound_pilot& pilot) {
+		assert((region_intersection(rr->incomplete_region, pilot.message.box) != pilot.message.box)
+		       == region_intersection(rr->incomplete_region, pilot.message.box).empty());
+		if(region_intersection(rr->incomplete_region, pilot.message.box) == pilot.message.box) {
+			handle_region_request_pilot(*rr, pilot, elem_size);
+			return true;
+		}
+		return false;
+	};
+	mrt->unassigned_pilots.erase(std::remove_if(mrt->unassigned_pilots.begin(), mrt->unassigned_pilots.end(), assign_pilot), mrt->unassigned_pilots.end());
+
+	return rr;
+}
+
+void receive_arbiter::begin_split_receive(
+    const transfer_id& trid, const region<3>& request, void* const allocation, const box<3>& allocated_box, const size_t elem_size) {
+	initiate_region_request(trid, request, allocation, allocated_box, elem_size);
+}
+
+async_event receive_arbiter::await_split_receive_subregion(const transfer_id& trid, const region<3>& subregion) {
+	// If there is no known associated `transfer`, we must have erased it previously due to the the entire `begin_split_receive` being completed. Any (partial)
+	// await thus immediately completes as well.
+	const auto transfer_it = m_transfers.find(trid);
+	if(transfer_it == m_transfers.end()) { return make_complete_event(); }
+
+	auto& mrt = std::get<multi_region_transfer>(transfer_it->second);
+
+#ifndef NDEBUG
+	// all boxes from the awaited region must be contained in a single allocation
+	const auto awaited_bounds = bounding_box(subregion);
+	assert(std::all_of(mrt.active_requests.begin(), mrt.active_requests.end(), [&](const stable_region_request& rr) {
+		const auto overlap = box_intersection(rr->allocated_box, awaited_bounds);
+		return overlap.empty() || overlap == awaited_bounds;
+	}));
+#endif
+
+	// If the transfer (by transfer_id) as a whole has not completed yet but the subregion is, this "await" also completes immediately.
+	const auto req_it = std::find_if(mrt.active_requests.begin(), mrt.active_requests.end(),
+	    [&](const stable_region_request& rr) { return !region_intersection(rr->incomplete_region, subregion).empty(); });
+	if(req_it == mrt.active_requests.end()) { return make_complete_event(); }
```

Comment:
This branch wasn't covered yet; I pushed a test case, as discussed in person!

Reply:
Thanks for the contribution!